.github/workflows/check-pull-request.yaml

```diff
@@ -29,9 +29,8 @@ jobs:
     strategy:
       matrix:
         go:
-        - 1.18
-        - 1.19
-        - '1.20'
+        - "1.20"
+        - "1.21"
```

Comment:
why was this added?

Reply:
I added 1.20 & 1.21 here since I seem to remember that there are projects on 1.20 & 1.21 e.g  CS & AMS respecitvely, so I thought we could test against both

Reply:
and why not bump the sdk to 1.21?

Reply:
I guess that will force any dependent project to upgrade too. Which is not too bad. It's mainly CS, AMS, OSL, AS, MSS, FM.

Reply:
Okay, let's give it a `go` to 1.21. 
At worst, dependent will be forced to update. Otherwise the go compiler will try to build the dependent package while ensuring the compatibleness promise.

Reply:
Updated

Reply:
> It's mainly CS, AMS, OSL, AS, MSS, FM.

@tzvatot @zgalor Just for information - there are other services using the ocm-sdk-go module e.g RHACS https://github.com/stackrox/acs-fleet-manager/blob/main/go.mod#L36

Reply:
When dependent components would like to upgrade to latest OCM SDK, then they will need to update go latest Go as well.

I think that's reasonable. Can you please also run it with BU?

Reply:
I've tagged BU on the JIRA.